features/dd-sdk-android-trace-otel/src/main/kotlin/com/datadog/android/trace/opentelemetry/internal/MiscUtils.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.trace.opentelemetry.internal
+
+import android.os.Build
+import com.datadog.android.api.InternalLogger
+
+internal const val NEEDS_DESUGARING_ERROR_MESSAGE =
+    "Trying to use OpenTelemetry SDK support for Android 23 and below. " +
+        "In order for this to properly work you will need to enable coreDesugaring " +
+        "in your compileOptions"
+
+internal fun <T : Any?> executeIfJavaFunctionPackageExists(
+    internalLogger: InternalLogger? = null,
+    defaultActionReturnValue: T,
+    action: () -> T
+): T {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        return action()
+    } else {
+        return try {
+            // we are forcing here the checkup as it could be triggered only at runtime and will be too late
+            @Suppress("UnsafeThirdPartyFunctionCall")
+            Class.forName("java.util.function.Function")
```

Comment:
are we sure that in case if desugaring applied there will be actually such class in the classpath (btw, if we use reflection, we need to add a rule to consumer Proguard rules) and desugaring won't instead inline function calls?

Reply:
I checked a desugared application and there is a `java.util.function` package added in the `.dex` package at compile time. I am not really sure we need proguard rule here. Everything we are doing here is quite experimental and we rely on the fact that the desugaring will always work like that.

Reply:
It is for obfuscation. Production app will have `java.util.function.Function` renamed.